ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/BlockAttestationsPool.java

```diff
@@ -0,0 +1,168 @@
+                      newAttestation.getData().getTarget().getEpoch(),
+                      dataRootToSlot);
+                }
+              },
+              () -> {
+                ChainStorage.add(newAttestation, aggregateAttesationsQueue);
+
+                ChainStorage.add(
+                    attestationDataHash,
+                    newAttestation.getAggregation_bits().copy(),
+                    this.unprocessedAttestationsBitlist);
+
+                ChainStorage.add(
+                    attestationDataHash,
+                    newAttestation.getData().getTarget().getEpoch(),
+                    dataRootToSlot);
+              });
+    }
+  }
+
+  public void addAttestationProcessedInBlock(Attestation attestation) {
+    Bytes32 attestationDataHash = attestation.getData().hash_tree_root();
+    ChainStorage.get(attestationDataHash, this.processedAttestationsBitlist)
+        .ifPresentOrElse(
+            oldBitlist -> {
+              for (int i = 0; i < attestation.getAggregation_bits().getCurrentSize(); i++) {
+                if (attestation.getAggregation_bits().getBit(i) == 1) oldBitlist.setBit(i);
+              }
+            },
+            () -> {
+              ChainStorage.add(
+                  attestationDataHash,
+                  attestation.getAggregation_bits().copy(),
+                  this.processedAttestationsBitlist);
+              ChainStorage.add(
+                  attestationDataHash,
+                  attestation.getData().getTarget().getEpoch(),
+                  dataRootToSlot);
+            });
+  }
+
+  public SSZList<Attestation> getAggregatedAttestationsForBlockAtSlot(UnsignedLong slot) {
+    SSZList<Attestation> attestations =
+        new SSZList<>(Attestation.class, Constants.MAX_ATTESTATIONS);
+    int numAttestations = 0;
+
+    while (aggregateAttesationsQueue.peek() != null
+        && aggregateAttesationsQueue.peek().getData().getSlot().compareTo(slot) <= 0
+        && numAttestations < Constants.MAX_ATTESTATIONS) {
+
+      Attestation aggregate = aggregateAttesationsQueue.remove();
+      Bytes32 attestationHashTreeRoot = aggregate.getData().hash_tree_root();
+
+      // Check if attestation has already been processed in successful block
+      if (ChainStorage.get(attestationHashTreeRoot, this.processedAttestationsBitlist)
+          .isPresent()) {
+        Bitlist bitlist =
+            ChainStorage.get(
+                    aggregate.getData().hash_tree_root(), this.processedAttestationsBitlist)
+                .get();
+        List<Integer> oldAttesters = getAttesterIndicesIntoCommittee(bitlist);
+        List<Integer> newAttesters =
+            getAttesterIndicesIntoCommittee(aggregate.getAggregation_bits());
+        if (oldAttesters.containsAll(newAttesters)) {
+          continue;
+        }
+      }
+      attestations.add(aggregate);
+      numAttestations++;
+    }
+    return attestations;
+  }
+
+  /* // Memory Cleaning
```

Comment:
Did you intentionally keep this commented out code?

Reply:
yes

Reply:
Added a jira ticket: https://pegasys1.atlassian.net/secure/RapidBoard.jspa?rapidView=67&projectKey=BC&view=planning&selectedIssue=BC-132